docs/src/userguide/device_offload.rst

```diff
@@ -0,0 +1,196 @@
+.. highlight:: cython
+
+.. _device_offload:
+
+********************************
+Offloading to a Device
+********************************
+
+Basic Usage
+===========
+
+Cython supports offloading code blocks to a GPU device.
+Currently only parallel blocks (see :ref:'Using Parallelism <parallel>')
+can be offloaded to a target device. The current implementation is based
+on the OpenMP 'target' directive.
+
+.. NOTE:: The offload functionality is experimental.
+
+.. NOTE:: Offloading to a device is only possible for code blocks which
+          do not require the Python interpreter. There is no way to use
+          Python objects on the device.
+
+Let's consider a simple example for computing pairwise distances between vectors in ``parallel``::
+
+    def pairwise_distance_host(double[:, ::1] X):
```

Comment:
I don't know if these ones are suitable, but if you include the (longer) code fragments from doc/examples then they get tested. This at least ensures the documentation is kind of true.

Reply:
Yep, I will make sure this gets done once I know how to enable offloading in the CI. Any suggestions for that will be appreciated.